react-native.config.js

```diff
@@ -1,85 +1,89 @@
-/**
- * This cli config is needed for the coexistance of react-native and other
- * out-of-tree implementations such react-native-macos.
- * The following issue is tracked by
- * https://github.com/react-native-community/discussions-and-proposals/issues/182
- *
- * The work-around involves having a metro.config.js for each out-of-tree
- * platform, i.e. metro.config.js for react-native and
- * metro.config.macos.js for react-native-macos.
- * This react-native.config.js looks for a --use-react-native-macos
- * switch and when present pushes --config=metro.config.macos.js
- * and specifies reactNativePath: 'node_modules/react-native-macos'.
- * The metro.config.js has to blacklist 'node_modules/react-native-macos',
- * and conversely metro.config.macos.js has to blacklist 'node_modules/react-native'.
- */
-'use strict';
-
+const fs = require('fs');
 const path = require('path');
 
-const dependencies = {
-  '@react-native-async-storage/async-storage': {
-    root: __dirname,
-    // rn-cli incorrectly resolves node_modules path for the following platforms
-    platforms: {
-      ios: null,
-      macos: null,
+const windowsProjectFile = path.join(
+  'node_modules',
+  '.generated',
+  'windows',
+  'ReactTestApp',
+  'ReactTestApp.vcxproj'
+);
+
+module.exports = {
+  dependencies: {
+    '@react-native-async-storage/async-storage': {
+      root: __dirname,
+      // rn-cli incorrectly resolves node_modules path for the following platforms
+      platforms: {
+        ios: null,
+        macos: null,
+      },
     },
-  },
-  // Suppress warnings about bob not being a proper native module
-  '@react-native-community/bob': {
-    platforms: {
-      android: null,
-      ios: null,
-      macos: null,
-      windows: null,
+    // Suppress warnings about bob not being a proper native module
+    '@react-native-community/bob': {
+      platforms: {
+        android: null,
+        ios: null,
+        macos: null,
+        windows: null,
+      },
     },
-  },
-  // We don't use Expo in our test apps
-  expo: {
-    platforms: {
-      android: null,
-      ios: null,
-      macos: null,
-      windows: null,
+    // We don't use Expo in our test apps
+    expo: {
+      platforms: {
+        android: null,
+        ios: null,
+        macos: null,
+        windows: null,
+      },
     },
   },
-};
+  project: {
+    android: {
+      sourceDir: path.join('example', 'android'),
+      manifestPath: path.relative(
+        path.join(__dirname, 'example', 'android'),
+        path.join(
+          path.dirname(require.resolve('react-native-test-app/package.json')),
+          'android',
+          'app',
+          'src',
+          'main',
+          'AndroidManifest.xml',
+        ),
+      ),
+    },
+    ios: {
```

Comment:
Because we specify options in `start:macos` script, we don't have to add this target here, right?